e2e/upgrade_test.go

```diff
@@ -140,13 +142,29 @@ var _ = Context("upgrade", func() {
 				if cond.Type != mocov1beta2.ConditionHealthy {
 					continue
 				}
-				if cond.Status == corev1.ConditionTrue {
+				if cond.Status == metav1.ConditionTrue {
 					return nil
 				}
 				return fmt.Errorf("cluster is not healthy: %s", cond.Status)
 			}
 			return errors.New("no health condition")
 		}).Should(Succeed())
+
+		Eventually(func() error {
+			cluster, err := getCluster("upgrade", "test")
+			if err != nil {
+				return err
+			}
+			ConditionUpToDate := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionUpToDate)
```

Comment:
This logic can be checked in the previous Eventually.
Instead, the status of StatefulSet should be checked here, not MySQLCluster.

Reply:
Do you mean checking healthy status of MySQLCluster after checking status of StatefulSet?

below is example code of my understanding.
```go
DELETE L136~L151
...
		Eventually(func() error {
			cluster, err := getCluster("upgrade", "test")
			if err != nil {
				return err
			}
			ConditionUpToDate := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionUpToDate)
			if ConditionUpToDate == nil {
				return fmt.Errorf("statefulset is not updated yet")
			}
			if ConditionUpToDate.Status != metav1.ConditionTrue {
				return fmt.Errorf("statefulset is not updated yet")
			}
			return nil
		}).Should(Succeed())
		ConditionHealthy := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionHealthy)
		Expect(ConditionHealthy).To(Equal(metav1.ConditionTrue))
...
```

Reply:
My expectation is as follows:

```go
		Eventually(func() error {
			cluster, err := getCluster("upgrade", "test")
			if err != nil {
				return err
			}
                        if cluster.metadata.generation != cluster.Status.reconcileInfo.generation {
                               return fmt.Errorf("cluster is not updated yet")
                        }
                        conditionHealthy := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionHealthy )
			if conditionHealthy == nil {
				return fmt.Errorf("cluster is not updated yet")
			}
			if conditionHealthy.Status != metav1.ConditionTrue {
				return fmt.Errorf("cluster is not updated yet")
			}
                        conditionUpToDate := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionUpToDate)
			if conditionUpToDate == nil {
				return fmt.Errorf("cluster  is not updated yet")
			}
			if conditionUpToDate.Status != metav1.ConditionTrue {
				return fmt.Errorf("cluster is not updated yet")
			}
			return errors.New("no health condition")
		}).Should(Succeed())
                var sts appsv1.StatefulSet
                err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.PrefixedName()}, &sts)
                Expect(err).Should(NotOccurred())
                Expect(sts.Status.CurrentRevision).Should(Equal(sts.Status.UpdateRevision))
                Expect(sts.Generation).Should(Equal(sts.Status.ObservedGeneration))
```

This test case is expected to detect a bug that Cluster status is updated before the StatefulSet is updated.
The logic in Eventually should be the same as ArgoCD health check logic.